internal/grpc/grpc.go

```diff
@@ -208,34 +209,83 @@ func GetDialOptions(agentConfig *config.Config, resourceID string) []grpc.DialOp
 
 	opts = append(opts, sendRecOpts...)
 
+	opts, skipToken := addTransportCredentials(agentConfig, opts)
+
+	if agentConfig.Command.Auth != nil && !skipToken {
+		opts = addPerRPCCredentials(agentConfig, resourceID, opts)
+	}
+
+	return opts
+}
+
+func addTransportCredentials(agentConfig *config.Config, opts []grpc.DialOption) ([]grpc.DialOption, bool) {
 	transportCredentials, err := getTransportCredentials(agentConfig)
-	if err == nil {
-		slog.Debug("Adding transport credentials to gRPC dial options")
-		opts = append(opts,
-			grpc.WithTransportCredentials(transportCredentials),
-		)
-	} else {
+	if err != nil {
+		slog.Error("Unable to add transport credentials to gRPC dial options", "error", err)
```

Comment:
This error log is misleading since we are adding the default transport credentials. Maybe it should be ```Unable to get transport credentials from agent configuration, adding default transport credentials to gRPC dial options```